cscs-checks/apps/amber/amber_check.py

```diff
@@ -197,3 +141,7 @@ def set_num_tasks_cray_shasta(self):
                 # need to decrease to just 8 nodes
                 self.num_nodes = 8
             self.num_tasks = self.num_nodes * self.num_tasks_per_node
+
+    @run_after('setup')
+    def set_reference_dict(self):
+        self.REFERENCE_DICT = REFERENCE_CPU_PERFORMANCE[self.scale]
```

Comment:
```suggestion
    @run_before('performance')
    def set_perf_reference(self):
        for key, val in REFERENCE_CPU_PERFORMANCE[self.scale].items():
            if self.current_partition.fullname in key:
                self.reference = {'*': val}
                break
        else:
            raise ValueError(
                f'could not find a reference for the current '
                f'partition {self.current_partition.fullname!r}'
            )
```

Reply:
Note that the keys of `REFERENCE_CPU_PERFORMANCE_LARGE` also must be made tuples in order for this to work.

Reply:
Why? Thank to string `if self.current_partition.fullname in key:` we check either substring in the string (as in case `REFERENCE_CPU_PERFORMANCE_LARGE`) or element in a tuple